Yosemite/Yosemite/Model/Mocks/ActionHandlers/MockOrderActionHandler.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+import Storage
+
+struct MockOrderActionHandler: MockActionHandler {
+
+    typealias ActionType = OrderAction
+
+    let objectGraph: MockObjectGraph
+    let storageManager: StorageManagerType
+
+    func handle(action: ActionType) {
+        switch action {
+            case .fetchFilteredAndAllOrders(let siteID, _, _, _, _, let onCompletion):
+                fetchFilteredAndAllOrders(siteID: siteID, onCompletion: onCompletion)
+            case .countProcessingOrders(let siteID, let onCompletion):
+                countOrders(status: .processing, siteID: siteID, onCompletion: onCompletion)
+
+            default: unimplementedAction(action: action)
+        }
+    }
+
+    func fetchFilteredAndAllOrders(siteID: Int64, onCompletion: @escaping (Error?) -> ()) {
+        saveOrders(orders: objectGraph.orders(forSiteId: siteID)) {
+            onCompletion(nil)
+        }
+    }
+
+    func countOrders(status: OrderStatusEnum, siteID: Int64, onCompletion: (OrderCount?, Error?) -> Void) {
+        let count = objectGraph.orders(withStatus: status, forSiteId: siteID).count
+
+        onCompletion(OrderCount(siteID: siteID, items: [
+            OrderCountItem(slug: "", name: "", total: count)
+        ]), nil)
+    }
+
+    private func saveOrders(orders: [Order], onCompletion: @escaping () -> ()) {
+        let storage = storageManager.newDerivedStorage()
+
+        storage.perform {
+            let updater = OrdersUpsertUseCase(storage: storage)
+            updater.upsert(orders)
+        }
+
+        storageManager.saveDerivedType(derivedStorage: storage) {
+            DispatchQueue.main.async(execute: onCompletion)
+        }
```

Comment:
nit: felt off-balance that we needed to dispatch the completion to main here but not in the other cases. Wondering if it wouldn't be better to make `saveDerivedType` dispatch its own completion on main itself 🤔 

Probably not a big deal; but this surprised me when reading the code during review and I can see this dispatch easily be missed in other call sites when people don't realize this… 🤷

Reply:
Yeah, unfortunately that's code outside that scope of this PR. 

IMHO we should never emit changes on the main queue from a library function – the app layer should handle when it wants to bring the changes into the main queue to avoid random UI hitches. 

But it's [the pattern used elsewhere in this app](https://github.com/woocommerce/woocommerce-ios/blob/a1d8473bbaff62a6baa03e4f3a848dd0df4efb57/Yosemite/Yosemite/Stores/OrderStore.swift#L344-L346), which is why I copied it for this 🤷‍♂️